design.md

```diff
@@ -0,0 +1,74 @@
+Gameplay:
+
+JOINS:
+
+* User (Baylee) creates a room and joins in.
+* Baylee is places in waiting room. 
+* Baylee sends invite to others.
+* Amjith joins the room, also placed in WAITING room.
+* Princess joins the room, also placed in WAITING room.
+* Sempi joins the room, also placed in WAITING room.
+
+STARTS:
+
+* Baylee starts the game.
+* Everyone is moved to the game room.
+* Every user is asked to write a story.
+* Every user name gets a little asterisk next to it denoting we're waiting for that user to finish.
+* Princess finishes their story and submits. 
+* Princess is now in WAITING state. Asterisk is removed from Princess status.
+* Baylee and Amjith submit their stories. They both go to WAITING state.
+* Sempi goes inactive.
+* Baylee initiates next step. 
+* Amjith and Princess approve. 
+* Everyone moves to guessing state except Sempi.
+* Sempi is sent back to waiting room.
```

Comment:
In this case Sempi has a status of `waiting` and we have a status of `writing` ?

It sounds like here are two scenarios to get out of waiting:
If I'm `waiting` and nobody is in the room I can begin the game
If I'm `waiting` and people are in the room, I must be approved?

The server has the userinfo, which could provide context to the send update method to have personalized statuses.

Reply:
In the second scenario, how would we "add" the user?

We could add a field `waiting` with the list of waiting users. If we are playing a game, and somebody appears in waiting we can accept them (POST /room/NAME/user/{id}/accept maybe?) Which would change their status